### PR TITLE
KAFKA-17741: Cleanup code base for JDK 11

### PR DESCRIPTION
**Pull request by frankvicky**
JIRA: [KAFKA-17741](https://issues.apache.org/jira/browse/KAFKA-17741)
>We are planning to drop JDK 8. This is part of the upgrade from JDK 8 to JDK 11. You can see the full plan here: [KAFKA-12894](https://issues.apache.org/jira/browse/KAFKA-12894)
> 1. `KafkaCompletableFuture` has many override methods from java 9, and they all need `@Override`annotation
> 2. remove `IS_JAVA9_COMPATIBLE` and `IS_JAVA11_COMPATIBLE`

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@frankvicky please fix conflicts

**Comment by frankvicky** [comment]
Hi @chia7712 
I have fixed the conflicts, PTAL

File: clients/src/main/java/org/apache/kafka/common/utils/ByteBufferUnmapper.java [hunk 1]
```diff
@@ -87,10 +87,7 @@ public static void unmap(String resourceDescription, ByteBuffer buffer) throws I
     private static MethodHandle lookupUnmapMethodHandle() {
         final MethodHandles.Lookup lookup = lookup();
         try {
-            if (Java.IS_JAVA9_COMPATIBLE)
-                return unmapJava9(lookup);
-            else
-                return unmapJava7Or8(lookup);
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/utils/ByteBufferUnmapper.java:93]: please remove `unmapJava7Or8`

File: clients/src/main/java/org/apache/kafka/common/utils/Crc32C.java [hunk 2]
```diff
@@ -37,10 +37,7 @@ public final class Crc32C {
     private static final ChecksumFactory CHECKSUM_FACTORY;
 
     static {
-        if (Java.IS_JAVA9_COMPATIBLE)
-            CHECKSUM_FACTORY = new Java9ChecksumFactory();
-        else
-            CHECKSUM_FACTORY = new PureJavaChecksumFactory();
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/utils/Crc32C.java:43]: please remove PureJavaChecksumFactory

File: clients/src/main/java/org/apache/kafka/common/utils/Crc32C.java [hunk 3]
```diff
@@ -37,10 +37,7 @@ public final class Crc32C {
     private static final ChecksumFactory CHECKSUM_FACTORY;
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/utils/Crc32C.java]: `private static final ChecksumFactory CHECKSUM_FACTORY = new Java9ChecksumFactory();`

File: clients/src/main/java/org/apache/kafka/common/utils/Java.java [hunk 4]
```diff
@@ -36,10 +36,6 @@ static Version parseVersion(String versionString) {
         return new Version(majorVersion, minorVersion);
     }
 
-    // Having these as static final provides the best opportunity for compiler optimization
-    public static final boolean IS_JAVA9_COMPATIBLE = VERSION.isJava9Compatible();
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/utils/Java.java:40]: please remove `isJava9Compatible` and `isJava11Compatible`

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for your patch.

File: clients/src/main/java/org/apache/kafka/common/utils/ByteBufferUnmapper.java [hunk 5]
```diff
@@ -87,40 +82,13 @@ public static void unmap(String resourceDescription, ByteBuffer buffer) throws I
...
-        Class<?> cleanerClass = directBufferCleanerMethod.type().returnType();
-        MethodHandle cleanMethod = lookup.findVirtual(cleanerClass, "clean", methodType(void.class));
-        MethodHandle nonNullTest = lookup.findStatic(ByteBufferUnmapper.class, "nonNull",
-                methodType(boolean.class, Object.class)).asType(methodType(boolean.class, cleanerClass));
-        MethodHandle noop = dropArguments(constant(Void.class, null).asType(methodType(void.class)), 0, cleanerClass);
-        return filterReturnValue(directBufferCleanerMethod, guardWithTest(nonNullTest, cleanMethod, noop))
-                .asType(methodType(void.class, ByteBuffer.class));
-    }
-
     private static MethodHandle unmapJava9(MethodHandles.Lookup lookup) throws ReflectiveOperationException {
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/utils/ByteBufferUnmapper.java]: Could you please inline this method?

File: clients/src/main/java/org/apache/kafka/common/utils/Java.java (hunk 4 above)
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/utils/Java.java:40]: Please remove `VERSION` also

File: clients/src/main/java/org/apache/kafka/common/utils/Crc32C.java [hunk 6]
```diff
@@ -34,14 +34,7 @@
...
 
-    private static final ChecksumFactory CHECKSUM_FACTORY;
-
-    static {
-        if (Java.IS_JAVA9_COMPATIBLE)
-            CHECKSUM_FACTORY = new Java9ChecksumFactory();
-        else
-            CHECKSUM_FACTORY = new PureJavaChecksumFactory();
-    }
+    private static final ChecksumFactory CHECKSUM_FACTORY = new Java9ChecksumFactory();
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/utils/Crc32C.java]: Please inline `Java9ChecksumFactory`

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for this cleanup

File: clients/src/main/java/org/apache/kafka/common/utils/Java.java [hunk 7]
```diff
@@ -22,8 +22,6 @@ public final class Java {
 
     private Java() { }
 
-    private static final Version VERSION = parseVersion(System.getProperty("java.specification.version"));
-
     // Package private for testing
     static Version parseVersion(String versionString) {
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/utils/Java.java]: Please remove unused method

Comment (chia7712) [reply clients/src/main/java/org/apache/kafka/common/utils/Java.java]: in `Java` file, only `isIbmJdk` and `isIbmJdkSemeru` are used

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for this patch

**Review by chia7712 — APPROVED** [review]

**Pull request merged**